src/adapter/src/coord/timestamp_oracle/postgres_oracle.rs

```diff
@@ -204,6 +204,60 @@ where
     async fn get_connection(&self) -> Result<Object, PoolError> {
         self.postgres_client.get_connection().await
     }
+
+    /// Returns a `Vec` of all known timelines along with their current greatest
+    /// timestamp (max of read_ts and write_ts).
+    ///
+    /// For use when initializing another [`TimestampOracle`] implementation
+    /// from another oracle's state.
+    ///
+    /// NOTE: This method can have linearizability violations, but it's hard to
+    /// get around those. They will only occur during bootstrap, and only
+    /// if/when we migrate between different oracle implementations. Once we
+    /// migrated from the catalog-backed oracle to the postgres/crdb-backed
+    /// oracle we can remove this code and it's callsite.
+    pub(crate) async fn get_all_timelines(
+        mut config: PostgresTimestampOracleConfig,
+    ) -> Result<Vec<(String, Timestamp)>, anyhow::Error> {
+        // Inject a dummy MetricsRegistry so the temporary PostgresClient
+        // doesn't clobber our regular metrics.
+        config.metrics_registry = MetricsRegistry::new();
+        let postgres_client = PostgresClient::open(config.into())?;
+
+        let client = postgres_client.get_connection().await?;
+
+        let q = r#"
+            SELECT EXISTS (SELECT * FROM information_schema.tables where table_name = 'timestamp_oracle');
+        "#;
+        let statement = client.prepare(q).await?;
+        let exists_row = client.query_one(&statement, &[]).await?;
+        let exists: bool = exists_row.try_get("exists").expect("missing exists column");
+        if !exists {
+            return Ok(Vec::new());
+        }
+
+        let q = r#"
+            SELECT timeline, GREATEST(read_ts, write_ts) as ts FROM timestamp_oracle;
+        "#;
+        let statement = client.prepare(q).await?;
+        let rows = client
+            .query(&statement, &[])
+            .await
+            .expect("todo: add internal retries");
```

Comment:
Is it worth doing this in a transaction with the previous statement? I can't think of a scenario off the top of my head where someone would delete the `timestamp_oracle` table, but I don't think it will be overly expensive.

Reply:
will do!